File: src-all/Rust/S0447-number-of-boomerangs.rs
/*
Given n points in the plane that are all pairwise distinct, a &quot;boomerang&quot; is a tuple of points (i, j, k) such that the distance between i and j equals the distance between i and k (the order of the tuple matters).

Find the number of boomerangs. You may assume that n will be at most 500 and coordinates of points are all in the range [-10000, 10000] (inclusive).

Example:

Input:
[[0,0],[1,0],[2,0]]

Output:
2

Explanation:
The two boomerangs are [[1,0],[0,0],[2,0]] and [[1,0],[2,0],[0,0]]

 

*/
pub struct Solution {}
impl Solution {
    pub fn number_of_boomerangs(points: Vec<Vec<i32>>) -> i32 {
        
    }
}

fn main() {
  assert_eq!(0, Solution::number_of_boomerangs(0));
  println!("Pass test cases!");
}

File: src-all/C++/S0447-number-of-boomerangs.cpp
/*
You are given n points in the plane that are all distinct, where points[i] = [xi, yi]. A boomerang is a tuple of points (i, j, k) such that the distance between i and j equals the distance between i and k (the order of the tuple matters).

Return the number of boomerangs.

 
Example 1:
Input: points = [[0,0],[1,0],[2,0]]
Output: 2
Explanation: The two boomerangs are [[1,0],[0,0],[2,0]] and [[1,0],[2,0],[0,0]].

Example 2:
Input: points = [[1,1],[2,2],[3,3]]
Output: 2

Example 3:
Input: points = [[1,1]]
Output: 0

 
Constraints:
	n == points.length
	1 <= n <= 500
	points[i].length == 2
	-104 <= xi, yi <= 104
	All the points are unique.


*/
#include <cassert>
#include <iostream>

class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        
    }
};

int main() {
  Solution sol;
  assert(sol.numberOfBoomerangs(0) == 0);
  std::cout << "Pass test cases!" << std::endl;
}
